base/menus/ButtonSelectMenu.js

```diff
@@ -0,0 +1,232 @@
+        this.game = game;
+        this.data = data;
+        this.buttons_keys = buttons;
+        this.titles = titles;
+        this.on_cancel = callbacks.on_cancel;
+        this.on_press = callbacks.on_press;
+        this.control_manager = control_manager;
+        this.buttons_number = buttons.length;
+        this.custom_scale = null;
+
+        const max_title_width = get_text_width(this.game, _.maxBy(titles, title => title.length));
+        this.title_window_width = title_window_width !== undefined ? title_window_width : max_title_width + 2 * (numbers.WINDOW_PADDING_H + numbers.INSIDE_BORDER_WIDTH);
+        const total_width = BUTTON_WIDTH * this.buttons_number + this.title_window_width + 2 * numbers.OUTSIDE_BORDER_WIDTH + 2;
+        this.dock_right = dock_right;
+
+        this.x = numbers.GAME_WIDTH - total_width;
+        if (!this.dock_right) {
+            this.x = this.x >> 1;
+        }
+        this.y = BUTTON_Y;
+
+        this.title_window = new Window(this.game, this.x + BUTTON_WIDTH * this.buttons_number, this.y, this.title_window_width, TITLE_WINDOW_HEIGHT);
+        this.group = game.add.group();
+        this.group.alpha = 0;
+        this.mount_buttons();
+        this.selected_button_index = 0;
+        this.menu_open = false;
+        this.menu_active = false;
+        this.group.width = 0;
+        this.group.height = 0;
+        this.selected_button_tween = null;
+    }
+
+    set_control() {
+        if(this.control_manager.initialized) this.control_manager.reset();
+        this.control_manager.set_control(true, false, true, false, {right: this.next_button.bind(this), left: this.previous_button.bind(this),
+            esc: this.on_cancel.bind(this), enter: this.on_press.bind(this)});
+    }
+
+    mount_buttons(filtered_buttons = []) {
+        const buttons = this.buttons_keys.filter(key => !filtered_buttons.includes(key));
+        this.buttons_number = buttons.length;
+        const total_width = BUTTON_WIDTH * this.buttons_number + this.title_window_width + 2 * numbers.OUTSIDE_BORDER_WIDTH + 2;
+        this.x = numbers.GAME_WIDTH - total_width;
+        if (!this.dock_right) {
+            this.x = this.x >> 1;
+        }
+        this.title_window.update_position({x: this.x + BUTTON_WIDTH * this.buttons_number});
+        if (this.buttons) {
+            this.buttons.forEach(obj => {
+                obj.sprite.destroy();
+            });
+        }
+        this.buttons = new Array(this.buttons_number);
+        for (let i = 0; i < this.buttons_number; ++i) {
+            this.buttons[i] = {
+                sprite: this.group.create(0, 0, "buttons", buttons[i]),
+                title: this.titles[i]
+            }
+            this.buttons[i].sprite.anchor.setTo(0.5, 1);
+            this.buttons[i].sprite.centerX = parseInt(BUTTON_WIDTH * (i + 0.5));
+            this.buttons[i].sprite.centerY = parseInt(BUTTON_HEIGHT * 0.5);
```

Comment:
Same thing here, I prefer `BUTTON_HEIGHT  >> 1`. But you can let it in the way you find it's ok.